### PR TITLE
Feature/pre commit

### PR DESCRIPTION
**Pull request by jonathangreen**
## Description

Switch our linters to run with [pre-commit](https://pre-commit.com/) instead of tox. That way you can setup a pre-commit hook and be notified when the code you are committing fails to lint. I did some testing with it today, and I found the pre-commit hook to work really nicely.

Broke it up into a couple commits. One adds the config and CI, one updates the readme and the final one contains all the automatic reformatting from the linters. The automatic reformatting shouldn't need much checking over, I manually reviewed it, and it seems harmless and improves readability.

## Motivation and Context

I was getting annoyed that I kept forgetting to lint the files I added.

## How Has This Been Tested?

Ran tests locally and with CI.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-json
+      - id: check-added-large-files
+      - id: check-ast
+      - id: mixed-line-ending
+      - id: requirements-txt-fixer
+        files: requirements.*\.txt
```
Comment (tdilauro) [inline .pre-commit-config.yaml:16]: Any reason not to include all of the checks in the [corresponding changes](https://github.com/ThePalaceProject/circulation-core/pull/47/commits/d66257389c0f06da9e3141e7d20277ec45d3ee85#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9) to `core`? And vice versa?

File: README.md [hunk 2]
```diff
@@ -145,6 +149,45 @@ Github Actions handles generating the `.rst` source files, generating the HTML s
...
+Pre-commit ships with a [number of lints](https://pre-commit.com/hooks.html) out of the box, we are configured to use:
+- `trailing-whitespace` - trims trailing whitespace.
+- `end-of-file-fixer` - ensures that a file is either empty, or ends with one newline.
+- `check-yaml` - checks yaml files for parseable syntax.
+- `check-json` - checks json files for parseable syntax.
+- `check-ast` - simply checks whether the files parse as valid python.
+- `check-shebang-scripts-are-executable` - ensures that (non-binary) files with a shebang are executable.
+- `check-merge-conflict` - checks for files that contain merge conflict strings.
+- `check-added-large-files` - prevents giant files from being committed.
+- `mixed-line-ending` - replaces or checks mixed line ending.
```
Comment (tdilauro) [inline README.md:179]: It doesn't look like `check-merge-conflict` is actually configured.

**Review by tdilauro — COMMENTED** [review]
Looks good, aside from a slight mismatch between the docs and config and whatever has suddenly triggered CodeQL.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jonathangreen) [reply .pre-commit-config.yaml:18]: Nope! Good catch. I thought i had. They should be synced up now.

File: README.md (hunk 2 above)
Comment (jonathangreen) [reply README.md:180]: Is now!

**Comment by jonathangreen** [comment]
@tdilauro Addressed those couple things. I think we can just ignore codeql. Those alerts were existing, but because this PR touched them, its tagging them on this PR. I have another task sometime to go through the existing alerts and triage them.

**Review by tdilauro — APPROVED** [review]
Good to go! 💃🏽

**Pull request merged**